File: app/java/src/main/java/com/example/production/domain/model/plan/Requirement.java
package com.example.production.domain.model.plan;

import lombok.*;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Requirement {
    private Integer id;
    private String requirementNumber;
    private Integer orderId;
    private String itemCode;
    private LocalDate dueDate;
    private BigDecimal requiredQuantity;
    @Builder.Default
    private BigDecimal allocatedQuantity = BigDecimal.ZERO;
    @Builder.Default
    private BigDecimal shortageQuantity = BigDecimal.ZERO;
    private String locationCode;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: app/csharp/src/ProductionManagement.Domain/Models/Plan/Requirement.cs
namespace ProductionManagement.Domain.Models.Plan;

/// <summary>
/// 所要情報
/// </summary>
public class Requirement
{
    public int Id { get; set; }

    public required string RequirementNumber { get; set; }

    public int OrderId { get; set; }

    public required string ItemCode { get; set; }

    public DateOnly DueDate { get; set; }

    public decimal RequiredQuantity { get; set; }

    public decimal AllocatedQuantity { get; set; } = 0m;

    public decimal ShortageQuantity { get; set; } = 0m;

    public required string LocationCode { get; set; }

    public DateTime CreatedAt { get; set; }

    public DateTime UpdatedAt { get; set; }

    // リレーション
    public Order? Order { get; set; }
    public Item.Item? Item { get; set; }
    public IList<Allocation>? Allocations { get; set; }
}
